execution-engine/engine-tests/src/test/hdac_system_contracts/genesis.rs

```diff
@@ -208,3 +263,303 @@ fn should_fail_if_bad_pos_install_contract_is_provided() {
 
     builder.run_genesis(&genesis_config);
 }
+
+#[ignore]
+#[should_panic]
+#[test]
+fn should_fail_delegate_is_more_than_stake() {
+    let account_1_balance = Motes::new(ACCOUNT_1_BALANCE.into());
+    let account_1 = {
+        let account_1_public_key = ACCOUNT_1_ADDR;
+        let account_1_bonded_amount = Motes::new(ACCOUNT_1_BONDED_AMOUNT.into());
+        GenesisAccount::new(
+            account_1_public_key,
+            account_1_balance,
+            account_1_bonded_amount,
+        )
+    };
+
+    let account_2_balance = Motes::new(ACCOUNT_2_BALANCE.into());
+    let account_2 = {
+        let account_2_public_key = ACCOUNT_2_ADDR;
+        let account_2_bonded_amount = Motes::new(ACCOUNT_2_BONDED_AMOUNT.into());
+        GenesisAccount::new(
+            account_2_public_key,
+            account_2_balance,
+            account_2_bonded_amount,
+        )
+    };
+
+    let state_infos = vec![
+        format_args!(
+            "d_{}_{}_{}",
+            base16::encode_lower(&ACCOUNT_1_ADDR.as_bytes()),
+            base16::encode_lower(&ACCOUNT_1_ADDR.as_bytes()),
+            ACCOUNT_1_BONDED_AMOUNT.to_string()
+        )
+        .to_string(),
+        format_args!(
+            "d_{}_{}_{}",
+            base16::encode_lower(&ACCOUNT_2_ADDR.as_bytes()),
+            base16::encode_lower(&ACCOUNT_2_ADDR.as_bytes()),
+            2_000_001.to_string()
+        )
+        .to_string(),
+    ];
+
+    let name = CHAIN_NAME.to_string();
+    let mint_installer_bytes = utils::read_wasm_file_bytes(MINT_INSTALL);
+    let pos_installer_bytes = utils::read_wasm_file_bytes(POS_INSTALL);
+    let standard_payment_installer_bytes = utils::read_wasm_file_bytes(STANDARD_PAYMENT_INSTALL);
+    let accounts = vec![account_1, account_2];
+    let protocol_version = ProtocolVersion::V1_0_0;
+    let wasm_costs = *DEFAULT_WASM_COSTS;
+
+    let genesis_config = GenesisConfig::new(
+        name,
+        TIMESTAMP,
+        protocol_version,
+        mint_installer_bytes,
+        pos_installer_bytes,
+        standard_payment_installer_bytes,
+        accounts,
+        state_infos,
+        wasm_costs,
+    );
```

Comment:
Couldn't it be replaced with `create_genesis_config`?

Reply:
https://github.com/hdac-io/CasperLabs/pull/22/commits/5ad6b1ca03393485276ce28f9d7c412edfd26475